Source/WebKit/WebProcess/Network/WebSocketChannel.cpp

```diff
@@ -147,14 +148,15 @@ WebSocketChannel::ConnectStatus WebSocketChannel::connect(const URL& url, const
     std::optional<FrameIdentifier> frameID;
     std::optional<PageIdentifier> pageID;
     StoredCredentialsPolicy storedCredentialsPolicy { StoredCredentialsPolicy::Use };
-    if (auto* frame = m_document ? m_document->frame() : nullptr) {
-        RefPtr mainFrame = m_document->localMainFrame();
+    document = m_document.get();
+    if (RefPtr frame = document ? document->frame() : nullptr) {
+        RefPtr mainFrame = document->localMainFrame();
         if (!mainFrame)
             return ConnectStatus::KO; 
         frameID = mainFrame->frameID();
         pageID = mainFrame->pageID();
-        if (auto* mainFrameDocumentLoader = mainFrame->document() ? mainFrame->document()->loader() : nullptr) {
-            auto* policySourceDocumentLoader = mainFrameDocumentLoader;
+        if (RefPtr mainFrameDocumentLoader = mainFrame->document() ? mainFrame->document()->loader() : nullptr) {
+            RefPtr policySourceDocumentLoader = mainFrameDocumentLoader;
```

Comment:
I don't really understand the use of this either. Also not in the old code.

Reply:
With old code, do you mean the code as it is on trunk?

Reply:
I think it was done this way since policySourceDocumentLoader can be a mainFrameDocumentLoader but also other values. I think policySourceDocumentLoader is the more important name here though so I think we can just drop the  mainFrameDocumentLoader variable, like I did in my latest upload.